include/pisa/cursor/block_max_scored_cursor.hpp

```diff
@@ -30,10 +30,12 @@ class BlockMaxScoredCursor: public MaxScoredCursor<Cursor> {
     ~BlockMaxScoredCursor() = default;
 
     [[nodiscard]] PISA_ALWAYSINLINE auto block_max_score() -> float { return m_wdata.score(); }
```

Comment:
@JMMackenzie Don't we need to multiply this one by the weight as well?

Reply:
I don't think so. If you have a look at the BMW algorithm you see, for example, the following:

```
              block_upper_bound +=
                    ordered_cursors[i]->block_max_score() * ordered_cursors[i]->query_weight();
```

So the `query_weight` component is being included here. I guess its up to interpretation as to whether this should be "silently" handled by propogating that weight back into the cursor itself (so then you do what you suggested and return m_wdata.score() * query_weight, and just use that single value in the algorithms).

Note that the `query_weight()` call is inherited from the `ScoredCursor` class, and this PR propagates (or at least, should propagate) the correct term weight back if `weighted` is set to true.

Hope this makes sense!

Reply:
Ah, yeah, I thought it was implemented somewhere but wasn't sure. I suppose it's a bit inconsistent but I don't want to refactor this now, seems like a lot of work. I'll leave a comment explaining it instead.

Reply:
Agreed, it's kind of weird. Once we merge here, we can make a new issue noting that a refactor is required. I could probably do it soon while this stuff is still fresh, but we'd need to consider whether there's ever a case where it would make sense to decouple the query term weight and the block weights during traversal.